### PR TITLE
Support negative `axis` for `F.softmax`

### PR DESCRIPTION
**Pull request by toslunar**
Merge #5496 first.

This PR also allows 1-dim inputs.

cf. #5381

### CODE REVIEW

**Comment by toslunar** [comment]
Jenkins, test this please.

**Comment by okuta** [comment]
LGTM!
